ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -71,44 +74,96 @@ class MainActivity : ComponentActivity() {
 @Composable
 fun WearApp(watchRepository: WatchRepository) {
     WearAppTheme {
-        // The state for the ScalingLazyColumn in other screen is hoisted to this level, so
-        // the Scaffold can properly place the position indicator (also known as the
-        // scroll indicator). We use it for various other things (like hiding the time when
-        // the user is scrolling).
-        //
-        // This current solution works for one scrolling type Composable, that is, we only
-        // have one Composable capable of scrolling.
-        val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+        // The state for any scrollable content in the composable screens is hoisted to this level,
+        // so the Scaffold can properly place the position indicator (also known as the
+        // scroll indicator).
+        // In our case, the watch list screen uses a ScalingLazyColumn and the watch details
+        // screen uses a Column with scrolling enabled.
+        // We also use the states for various other things (like hiding the time when the user is
+        // scrolling).
+        // ----------
+
+        // For all scrolling screens, we save their scrolling offset in a remembered map (as a
+        // ScrollableState), so when users return to the screen, they are at the same location as
+        // they left.
+        // We set the default to the watch list screen (ScalingLazyListState) since there is only
+        // one. For the watch detail screens (with many different variations based on the watch id),
+        // we add those dynamically to the map as they are needed (in the navigation section).
+        var currentScrollStateKey: String = remember { Screen.WatchList.route }
+        val scrollStates: MutableMap<String, ScrollableState> = remember {
+            mutableMapOf(currentScrollStateKey to ScalingLazyListState())
+        }
 
         val swipeDismissableNavController = rememberSwipeDismissableNavController()
 
-        // Allows user to disable the text before the time and hide the vignette.
+        // Allows user to disable the text before the time.
         var showProceedingTextBeforeTime by rememberSaveable { mutableStateOf(false) }
-        var vignetteVisible by rememberSaveable { mutableStateOf(true) }
+
+        // Allows user to show/hide the vignette on appropriate screens.
+        // IMPORTANT NOTE: Usually you want to show the vignette all the time on screens with
+        // scrolling content, a rolling side button, or a rotating bezel. This preference is just
+        // to visually demonstrate the vignette for the developer to see it on and off.
+        var vignetteVisiblePreference by rememberSaveable { mutableStateOf(true) }
```

Comment:
Can the Vignette be driven off the ScrollableState? e.g. enabled for ScalingLazyListState when we know there is enough content to cause scrolling?

Reply:
Hmmm... I suppose we could in a followup PR. You mean, if it is some sort of scrolling window BUT it doesn't have enough content to scroll, don't show the Vignette?

Reply:
Ignore, think we can do this outside the sample.  I'll discuss with you.